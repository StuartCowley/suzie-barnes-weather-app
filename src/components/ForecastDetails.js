import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const formattedDate = new Date(forecast.date).toDateString();
  //   console.log(temperature.min, "<== min temperature");
  // console.log(formattedDate);
  // console.log(forecast.date);
  return (
    <div className="forecast-details">
      <div className="forecast-details__date"> {formattedDate}</div>
      <div className="forecast-details__temperaturemax">
        Highs of {forecast.temperature.max}&deg;C
      </div>
      <div
        className="forecast-details__temperaturemin"
        data-testid="forecast-mintemperature"
      >
        Lows of {forecast.temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity: {forecast.humidity}%
      </div>
      <div className="forecast-details__windspeed">
        Wind Speed: {forecast.wind.speed}mph
      </div>
      <div className="forecast-details__winddirection">
        Wind Direction: {forecast.wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
  }).isRequired,
};

export default ForecastDetails;
