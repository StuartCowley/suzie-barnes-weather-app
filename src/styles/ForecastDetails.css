.forecast-details {
  display: flex;
  justify-content: center;
  background-color: rgb(255, 226, 64);
  border-radius: 10px;
  width: auto;
  height: auto;
  align-items: center;
}

.forecast-details__date {
  font-size: 20px;
  font-weight: bold;
}
