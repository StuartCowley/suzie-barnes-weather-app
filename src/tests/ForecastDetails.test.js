import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 20,
    wind: 10,
    temperature: {
      min: 12,
      max: 22,
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        humidity={validProps.humidity}
        wind={validProps.wind}
        temperature={validProps.temperature}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
