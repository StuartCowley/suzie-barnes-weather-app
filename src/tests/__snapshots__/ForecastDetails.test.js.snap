// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
       Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__temperaturemax"
    >
      22°C
    </div>
    <div
      class="forecast-details__temperaturemin"
      data-testid="forecast-mintemperature"
    >
      12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      20%
    </div>
    <div
      class="forecast-details__windspeed"
    >
      10mph
    </div>
    <div
      class="forecast-details__winddirection"
    >
      NE
    </div>
  </div>
</DocumentFragment>
`;
